src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -406,9 +399,19 @@ public void close() {
             resubscribeFuture.cancel(true);
         }
         try {
-            disconnect().get(getCloseTimeout(), TimeUnit.MILLISECONDS);
+            disconnect().get(getTimeout(), TimeUnit.MILLISECONDS);
         } catch (InterruptedException | ExecutionException | TimeoutException e) {
             logger.atError().log("Error while disconnecting the MQTT client", e);
         }
     }
+
+    public synchronized void closeOnShutdown() {
+        if (resubscribeFuture != null && !resubscribeFuture.isDone()) {
+            logger.atTrace().log("Canceling resubscribe future");
+            resubscribeFuture.cancel(true);
+        }
+        // During Nucleus shutdown, disconnecting AwsIotMqttClient would time out because SDK/CRT backend cannot
```

Comment:
this comment doesn't explain what you're doing.

The important part is that you aren't waiting for the disconnect future (or returning it).

Reply:
will update the comment